src/rest_vol_object.c

```diff
@@ -790,20 +819,26 @@ RV_object_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_object_s
 
```

Comment:
To be safe, you should initialize `iter_object->handle_path = NULL` here so that the added free call in the `done:` section below doesn't accidentally try to free a random pointer if some code is later added between the allocation of `iter_object` and the allocation/setting of `iter_object->handle_path`.